src/DependencyInjection/Compiler/PaginatorAwarePass.php

```diff
@@ -34,7 +34,7 @@ public function process(ContainerBuilder $container): void
     {
         $defaultAttributes = ['paginator' => 'knp_paginator'];
 
-        foreach ($container->findTaggedServiceIds(self::PAGINATOR_AWARE_TAG) as $id => $attributes) {
+        foreach ($container->findTaggedServiceIds(self::PAGINATOR_AWARE_TAG) as $id => [$attributes]) {
```

Comment:
This is actually interesting. Because of a misconfiguration of the `ContainerBuilder` mock, the tests did not catch that the attributes weren't merged correctly.